src/getVersion.c

```diff
@@ -15,8 +15,14 @@
 // handleGetVersion is the entry point for the getVersion command. It
 // unconditionally sends the app version.
 void handleGetVersion(uint8_t p1, uint8_t p2, uint8_t *dataBuffer, uint16_t dataLength, volatile unsigned int *flags, volatile unsigned int *tx) {
-	G_io_apdu_buffer[0] = APPVERSION[0] - '0';
-	G_io_apdu_buffer[1] = APPVERSION[2] - '0';
-	G_io_apdu_buffer[2] = APPVERSION[4] - '0';
-	io_exchange_with_code(SW_OK, 3);
+    // We return HTR before the app version, so our wallet can identify it's talking
+    // to the Hathor app. Otherwise, we could be making this request for another app
+    // and would also get a version from it.
+	G_io_apdu_buffer[0] = 'H';
```

Comment:
Is this the magic number? I think I've added a comment about it on the first PR. If it's this we can resolve it there

Reply:
Yes. I wanted to have a PR just for that, but we were taking so long to close these ones that I decided to place them here.